docs: Add a section in the Bun guide to use DHI (#23565)
docker/docker.github.io: content/guides/bun/containerize.md

│ └── README.md
```

## Create a Dockerfile

Before creating a Dockerfile, you need to choose a base image. You can use either the official image from Oven (the company behind Bun) or a Docker Hardened Image (DHI) from your private DHI catalog.

> The benefit of choosing DHI is that it is a security-focused image that is built to be minimal and secure. Learn more about DHI [here](/dhi/).

Reviewer: ```suggestion
Choosing DHI offers the advantage of a production-ready image that is lightweight and secure. For more information, see [Docker Hardened Images](https://docs.docker.com/dhi/).
```

CMD ["bun", "server.js"]
```

{{< /tab >}}
{{< tab name="Using Docker Hardened Images" >}}
Docker Hardened Images (DHIs) are available for Bun on [Docker Hub](https://hub.docker.com/hardened-images/catalog/dhi/bun). Unlike using the official image, you must first mirror the Bun image into your organization and then use it as your base image. Follow the [DHI quickstart](/dhi/get-started/) to create a mirrored repository for Bun.

Mirrored repositories must start with `dhi-`, for example: `FROM <your-namespace>/dhi-bun:<tag>`. In the following Dockerfile, the `FROM` instruction uses `<your-namespace>/dhi-bun:latest` as the base image.

```dockerfile
# Use the DHI Bun image as the base image

> FROM <your-namespace>/dhi-bun:latest

Reviewer 2: There is no `latest` tag for DHIs. It'll need a specific version.

CMD ["bun", "server.js"]
```

{{< /tab >}}
{{< /tabs >}}

> Aside from specifying the base image, the Dockerfile also:

Reviewer: ```suggestion
In addition to specifying the base image, the Dockerfile also:
```
